src/platforms/shared/setup/ad-engine-phases-setup.ts

```diff
@@ -0,0 +1,57 @@
+import {
+	adEnginePhases,
+	context,
+	DiProcess,
+	documentLoadedPromise,
+	domContentLoadedPromise,
+	eventsRepository,
+	InstantConfigService,
+	waitForEventPromise,
+} from '@wikia/ad-engine';
+import { Injectable } from '@wikia/dependency-injection';
+import { addExperimentGroupToTargeting } from '../utils/experiment-targeting';
+
+const DEFAULT_CONFIGURATION_PHASE_WAIT_TIMEOUT = 200;
+const DEFAULT_CONFIGURATION_PHASE_DELAY = 50;
+
+const DEFAULT_PARTNERS_PHASE_WAIT_TIMEOUT = 500;
+const DEFAULT_PARTNERS_PHASE_DELAY = 50;
+
+const DEFAULT_AD_CALL_WAIT_TIMEOUT = 2000;
+
+@Injectable()
+export class AdEnginePhasesSetup implements DiProcess {
+	constructor(protected instantConfig: InstantConfigService) {}
+
+	execute(): void {
+		this.setupWarpContext();
+		const isWarpEnabled = context.get('options.phases.enabled');
+
+		if (!isWarpEnabled) return;
+
+		addExperimentGroupToTargeting('warp');
+
+		adEnginePhases.enabled = isWarpEnabled;
+
+		adEnginePhases.configuration = domContentLoadedPromise(
+			context.get('options.phases.configurationTimeout') ??
+				DEFAULT_CONFIGURATION_PHASE_WAIT_TIMEOUT,
+			context.get('options.phases.configurationDelay') ?? DEFAULT_CONFIGURATION_PHASE_DELAY,
+		);
+
+		adEnginePhases.partners = documentLoadedPromise(
+			context.get('options.phases.partnersTimeout') ?? DEFAULT_PARTNERS_PHASE_WAIT_TIMEOUT,
+			context.get('options.phases.partnersDelay') ?? DEFAULT_PARTNERS_PHASE_DELAY,
+		);
+
+		adEnginePhases.firstAdCall = waitForEventPromise(
+			eventsRepository.AD_ENGINE_SLOT_LOADED,
+			context.get('options.phases.firstAdCallTimeout') ?? DEFAULT_AD_CALL_WAIT_TIMEOUT,
+		);
+	}
+
+	setupWarpContext(): void {
+		context.set('options.phases', this.instantConfig.get('icWarp', {}));
+		context.set('options.phases.enabled', window?.ads?.context?.warp);
```

Comment:
```suggestion
		context.set('options.phases.enabled', window?.ads?.context?.warp === true);
```
this way we will always deal with boolean (as JS handles logical expressions in a very weird way)